src/groups/mqb/mqbauthn/mqbauthn_anonauthenticator.h

```diff
@@ -0,0 +1,189 @@
+/// for testing or controlling anonymous access.
+/// @bbref{mqbauthn::AnonAuthenticatorPluginFactory} is the corresponding
+/// factory class for the authenticator plugin.
+
+// MQB
+#include <mqbplug_authenticator.h>
+
+// BDE
+#include <ball_log.h>
+#include <bsl_memory.h>
+#include <bsl_optional.h>
+#include <bsl_string.h>
+#include <bsl_string_view.h>
+#include <bsla_annotations.h>
+#include <bslma_managedptr.h>
+#include <bslma_usesbslmaallocator.h>
+#include <bsls_keyword.h>
+#include <bsls_types.h>
+
+namespace BloombergLP {
+
+// FORWARD DECLARATION
+namespace mqbcfg {
+class AuthenticatorPluginConfig;
+}
+
+namespace mqbauthn {
+
+// ==============================
+// class AnonAuthenticationResult
+// ==============================
+
+class AnonAuthenticationResult : public mqbplug::AuthenticationResult {
+  private:
+    // DATA
+    bsl::string                       d_principal;
+    bsl::optional<bsls::Types::Int64> d_lifetimeMs;
+
+  public:
+    // TRAITS
+    BSLMF_NESTED_TRAIT_DECLARATION(AnonAuthenticationResult,
+                                   bslma::UsesBslmaAllocator)
+
+    // CREATORS
+
+    /// Construct this object using the optionally specified `allocator`.
+    AnonAuthenticationResult(bsl::string_view                  principal,
+                             bsl::optional<bsls::Types::Int64> lifetimeMs,
+                             bslma::Allocator*                 allocator);
+
+    ~AnonAuthenticationResult() BSLS_KEYWORD_OVERRIDE;
+
+    // ACCESSORS
+
+    bsl::string_view principal() const BSLS_KEYWORD_OVERRIDE;
+    const bsl::optional<bsls::Types::Int64>&
+    lifetimeMs() const BSLS_KEYWORD_OVERRIDE;
+};
+
+// =======================
+// class AnonAuthenticator
+// =======================
+
+class AnonAuthenticator : public mqbplug::Authenticator {
+  public:
+    // PUBLIC CLASS DATA
+    static BSLS_KEYWORD_INLINE_CONSTEXPR bsl::string_view k_NAME =
+        "AnonAuthenticator";
+    static BSLS_KEYWORD_INLINE_CONSTEXPR bsl::string_view k_MECHANISM =
+        "ANONYMOUS";
+
+  private:
+    // CLASS-SCOPE CATEGORY
+    BALL_LOG_SET_CLASS_CATEGORY("MQBAUTHN.ANONAUTHENTICATOR");
+
+    // DATA
+    bslma::Allocator* d_allocator_p;
+
+    bool d_isStarted;
+
+    /// If true, authentication always succeeds; if false, always fails.
+    bool d_shouldPass;
+
+  private:
+    // NOT IMPLEMENTED
+    AnonAuthenticator(const AnonAuthenticator& other) BSLS_KEYWORD_DELETED;
+    AnonAuthenticator&
+    operator=(const AnonAuthenticator& other) BSLS_KEYWORD_DELETED;
+
+  public:
+    // TRAITS
+    BSLMF_NESTED_TRAIT_DECLARATION(AnonAuthenticator,
+                                   bslma::UsesBslmaAllocator)
+
+    // CREATORS
+
+    /// Construct an AnonAuthenticator with the specified `config`.  If
+    /// `shouldPass` in `config` is set to true, authentication will always
+    /// succeed; if false, authentication will always fail.  If no config is
+    /// set, authentication defaults to always succeed.
```

Comment:
```suggestion
    /// succeed; if false, authentication will always fail.  If no config is
    /// provided, or `shouldPass` setting is missing, authentication
    /// defaults to always succeed.
```
Might be good to mention NULL config case